x/profiles/spec/06_client.md

```diff
@@ -0,0 +1,710 @@
+        },
+        "clientId": "desmos16c60y8t8vra27zjg2arlcd58dck9cwn7p6fwtd-twitter-lucagraziotti"
+      },
+      "result": {
+        "success": {
+          "value": "6c7563616772617a696f747469",
+          "signature": "3558f0a50c68d48f66f808f7d12f90be07ffecf60bbdb67e514b3b00c0d30322644168a8369d991df65ff2f8e4890ab2ef5677a33a1304a1c1dff8e627296dde"
+        }
+      },
+      "creationTime": "2021-11-04T10:50:54.794047348Z"
+    }
+  ],
+  "pagination": {
+    "total": "1"
+  }
+}
+```
+
+### ApplicationLinkByClientID
+The `ApplicationLinkByClientID` endpoint allows users to query an application link given a client id. 
+
+```bash
+desmos.profiles.v2.Query/ApplicationLinkByClientID
+```
+
+Example:
+```bash
+grpcurl -plaintext \
+  -d '{"client_id": "desmos16c60y8t8vra27zjg2arlcd58dck9cwn7p6fwtd-twitter-lucagraziotti"}' localhost:9090 desmos.profiles.v2.Query/ApplicationLinkByClientID
+```
+
+Example Output: 
+```json
+{
+  "link": {
+    "user": "desmos16c60y8t8vra27zjg2arlcd58dck9cwn7p6fwtd",
+    "data": {
+      "application": "twitter",
+      "username": "lucagraziotti"
+    },
+    "state": "APPLICATION_LINK_STATE_VERIFICATION_SUCCESS",
+    "oracleRequest": {
+      "id": "537363",
+      "oracleScriptId": "32",
+      "callData": {
+        "application": "twitter",
+        "callData": "7b226d6574686f64223a227477656574222c2276616c7565223a2231343536323036353331383830343233343330227d"
+      },
+      "clientId": "desmos16c60y8t8vra27zjg2arlcd58dck9cwn7p6fwtd-twitter-lucagraziotti"
+    },
+    "result": {
+      "success": {
+        "value": "6c7563616772617a696f747469",
+        "signature": "3558f0a50c68d48f66f808f7d12f90be07ffecf60bbdb67e514b3b00c0d30322644168a8369d991df65ff2f8e4890ab2ef5677a33a1304a1c1dff8e627296dde"
+      }
+    },
+    "creationTime": "2021-11-04T10:50:54.794047348Z"
+  }
+}
+```
+
+### ApplicationLinkOwners
+The `ApplicationLinkOwners` endpoint allows users to query the application link owners given an optional application name and username. 
+
+```bash
+desmos.profiles.v2.Query/ApplicationLinkOwners
+```
+
+**Note**
+The `user` parameter will be used only if the `application` one is specified as well. 
+
+Example:
+```bash
+grpcurl -plaintext \
+  -d '{"application": "twitter", "user": "lucagraziotti"}' localhost:9090 desmos.profiles.v2.Query/ApplicationLinkOwners
+```
+
+Example Output:
+```json
+{
+  "owners": [
+    {
+      "user": "desmos16c60y8t8vra27zjg2arlcd58dck9cwn7p6fwtd",
+      "application": "twitter",
+      "username": "lucagraziotti"
+    }
+  ],
+  "pagination": {
+    "total": "1"
+  }
+}
+```
+
+### Params
+The `Params` endpoint allows users to query the current params of the `profiles` module. 
+
+```bash
+desmos.profiles.v2.Query/Params
+```
+
+Example:
+```bash
+grpcurl localhost:9090 desmos.profiles.v2.Query/Params
+```
+
+Example Output:
+```json
+{
+  "params": {
+    "nickname": {
+      "minLength": "Mg==",
```

Comment:
these lengths should be string but representing numbers: e.g. "3"

Reply:
This is how Protobuf returns bytes since we have set the params to be that type